### PR TITLE
HDDS-11069. Block location is missing in output of Ozone debug chunkinfo command for EC.

### PR DESCRIPTION
**Pull request by sadanand48**
## What changes were proposed in this pull request?
After HDDS-10983 , it requires every getBlock request be assigned a replicaIndex in case of an EC block. The request constructed for getBlock in this tool didn't have that. This change sets the replicaIndex if key is EC in the getBlock request.
I have reconstructed the request after unpacking the existing request which could have been avoided but since the usecase is just a debug tool and it was simple change, I went with it.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-11069

## How was this patch tested?
Unit test

### CODE REVIEW

**Comment by SaketaChalamchala** [comment]
@swamirishi could you please take a look?

File: hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientGrpc.java [hunk 1]
```diff
@@ -316,6 +317,37 @@ public ContainerCommandResponseProto sendCommand(
...
+          request.getGetBlock();
+      DatanodeBlockID datanodeBlockID = getBlockRequestProto.getBlockID();
+      DatanodeBlockID newDatanodeBlockID =
+          DatanodeBlockID.newBuilder(datanodeBlockID)
+              .setReplicaIndex(replicaIdx).build();
+      ContainerProtos.GetBlockRequestProto newGetBlockRequestProto =
+          ContainerProtos.GetBlockRequestProto.newBuilder(getBlockRequestProto)
+              .setBlockID(newDatanodeBlockID).build();
+      request = ContainerCommandRequestProto.newBuilder(request)
+          .setGetBlock(newGetBlockRequestProto).build();
```
Comment (nandakumar131) [inline hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientGrpc.java:346]:
```suggestion
       ContainerProtos.GetBlockRequestProto gbr = request.getGetBlock();
       request = request.toBuilder().setGetBlock(gbr.toBuilder().setBlockID(
            gbr.getBlockID().toBuilder().setReplicaIndex(
                pipeline.getReplicaIndex(dn)).build()).build()).build();
```

Comment (sadanand48) [reply hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientGrpc.java]: Thanks , updated.

**Review by aryangupta1998 — APPROVED** [review]
LGTM!

**Comment by aryangupta1998** [comment]
Thanks @sadanand48 for the patch, @nandakumar131 for the review.

**Pull request merged**